packages/wpcom-checkout/src/product-url-encoding.ts

```diff
@@ -0,0 +1,20 @@
+// Some product slugs or meta contain slashes which will break the URL, so we
+// encode them first. We cannot use encodeURIComponent because the calypso
+// router seems to break if the trailing part of the URL contains an encoded
+// slash (eg: /checkout/example.com/foo%2Fbar breaks routing). We cannot use
+// the same characters for encoding as for decoding if they are UTF-8 encoded
+// because they will already be decoded by the time the decoding takes place.
+//
+// If this is ever changed, please make sure to also change the code that
+// generates renewal emails on the backend, because it uses the same encoding!
+
+const slashForEncoding = '%25';
+const slashForDecoding = '%'; // WARNING: This will be used in a RegExp so it must not contain RegExp special characters unless they are escaped!
+
+export function encodeProductForUrl( slug: string ): string {
+	return slug.replace( /\//g, slashForEncoding );
+}
+
+export function decodeProductFromUrl( slug: string ): string {
+	return slug.replace( new RegExp( slashForDecoding, 'g' ), '/' );
```

Comment:
Worth considering doing this without `RegExp`?

`return slug.split( slashForDecoding ).join( '/' );`

Reply:
Hm... interesting. I'm torn. On the one hand, any chance to avoid using a RegExp is a good thing. On the other, I think that the split/join is slightly less readable. 🤔  Thanks for the suggestion. I might take it.

Reply:
`String.replaceAll` is on its way. I'm not sure if it's something that could work now depending on how the build it set up.

Reply:
Yeah, that was what I wanted but I think it's not supported enough yet. I added unit tests and that gave me enough confidence to use your suggestion.